src/Nethermind/Nethermind.Init/Steps/InitializePrecompiles.cs

```diff
@@ -30,7 +31,9 @@ public async Task Execute(CancellationToken cancellationToken)
 
             try
             {
+                if (_wasSetup) return;
```

Comment:
maybe better to have some kind of static lock here?

Reply:
switched to `Interlocked`

Reply:
might have made it worse